### ISSUE TITLE
Term enrichment through AmiGO errors out on too much input (500 for overloading forward)

### ISSUE THREAD

**Issue by kltm** [geneontology/amigo#198]
Due to the changes necessary to forward to PANTHER in #171, very large inputs can cause the header to be incomplete when forwarding, causing a 500 error.

```
 Error message:
malformed header from script. Bad header=0ADDB_G0291952%0ADDB_G0272781%: rte 
```

This particular issue will be considered fixed with either:

* a catch of the error, an explanation, and an offer to send the user to PANTHER
* a count of the input on the AmiGO side with an arbitrary and reported limit

It should be noted that there is no way around this issue completely until we no longer forward--forwards cannot contain POSTy data, and large GETs can get truncated, sometimes destroying the header (as in this case).

We got this from: http://jira.geneontology.org/browse/GO-841

**Comment by kltm** [comment]
@cmungall Another question to answer is which is worse: the lack of interface offered by #171 or this issue? I'm beginning to think that reverting with a better explanation may be preferable to the current state.

**Comment by kltm** [comment]
Okay, taking a more serious look at this again, I now remember the trade-offs that were made to get here. IIRC, forwards were chosen to preserve the bulk of the RTE/TERP mechanism in AmiGO, without realizing the typical size profile of users.
If we abandon all of the previous mechanism, we can probably do cross posting fine (if I remeber about the PANTHER API correctly--will check on that next).

**Comment by paolaroncaglia** [comment]
Just wondering if there are any updates on this please? We keep getting GO helpdesk queries about that error :-}

Thanks.

**Comment by kltm** [comment]
Unfortunately, there has been very little development time to be given to AmiGO recently. All things being equal, "fixing" #171 was probably a bad idea.

**Comment by mcourtot** [comment]
It would be great to add a hyperlink towards the FAQ so users running into the issue may not have to email the helpdesk and wait for answers, see #223

**Comment by kltm** [comment]
I believe I have a possible fix in for the main irritants here. If somebody else can confirm that this works, I'll ping production to get it out:

 http://nakama.berkeleybop.org (2.2.x)
 http://tomodachi.berkeleybop.org (2.3.x)

**Comment by mcourtot** [comment]
I tried with the list we received yesterday at http://jira.geneontology.org/browse/GO-921 (list is in body of text) and it seems to work (I tried both servers). Pinging @paolaroncaglia in case she has other lists from users in her inbox for additional test.

**Comment by kltm** [comment]
Excellent--if it works out, we'll cut a quick release before the meeting.

**Comment by paolaroncaglia** [comment]
I tried a list of ~650 genes (from an old paper of mine, http://www.ncbi.nlm.nih.gov/pubmed/19887551) on both servers and it worked fine. More relevantly, I tried the expanded list of ~2000 genes from the same study (i.e. without fold-change cut-off) and it also worked, on the nakama server though. The tomodachi server resulted in an empty result page, showing that all submitted IDs are unmapped. But I tried a second time and it worked. 

Super!!

Thanks,
Paola
